common/pldm/src/codec.rs

```diff
@@ -0,0 +1,36 @@
+// Licensed under the Apache-2.0 license
+
+#[derive(Debug)]
+pub enum PldmCodecError {
+    BufferTooShort,
+    Unsupported,
+}
+
+/// A trait for encoding and decoding PLDM (Platform Level Data Model) messages.
+///
+/// This trait provides methods for encoding a PLDM message into a byte buffer
+/// and decoding a PLDM message from a byte buffer. Implementers of this trait
+/// must also implement the `Debug` trait and be `Sized`.
+pub trait PldmCodec: core::fmt::Debug + Sized {
```

Comment:
I think we could remove a lot of the `impl` code below if we provide a default `impl` for `FromBytes + IntoBytes`, since the code is identical for all of those (they simply call `write_to` and `read_from`), and only make custom `impl`s when needed (for things that can't use `zerocopy`).